ui/src/app/utils/stores.tsx

```diff
@@ -29,20 +29,27 @@ export const cdrVersionStore = atom<CdrVersionTiersResponse>({tiers: []});
 
 export interface ProfileStore {
   profile?: Profile;
+  load: Function;
   reload: Function;
   updateCache: Function;
 }
 
 export const profileStore = atom<ProfileStore>({
   profile: null,
+  load: async() => {
+    if (!profileStore.get().profile) {
```

Comment:
Note: as a general pattern, ideally this holds some kind of state, or else memoizes the call. Two callers of `load` hitting this simultaneously will generate two requests.

This may be in broader in scope, but also looking at the overall API here, there's definitely an argument to remove synchronous `.profile` access and put a Promise in its place. This would make access patterns consistent and mean callers don't need to bake in expectations around internal lifecycle details or sprinkle in null checks. It's also fine to leave synchronous access if we want to treat profile as "required data to render the app".

Reply:
I've been futzing with blocking on an existing request, and to do it cleanly actually requires returning a promise from these functions anyways, so I may as well make everything in the store into a promise...

Reply:
Making `profile` a promise would require changing how our react route guards work which is making my brain hurt so perhaps I will try this again tomorrow while encoffeed

Reply:
I agree on memoizing the call. On promisifying the profile, I think once the angular code is removed we will have some better options for dealing with the profile. We could wrap the app in something like a `withProfile` HOC at a high level, or put a guard in place.

Reply:
I'm not gonna be able to make the profile a Promise without massively blowing up the scope of this PR. I can certainly make it so there's no more than three load calls (AppComponent, DisabledGuard, RegistrationGuard) and even in that cases DisabledGuard should be the first and only one making that call because AppComponent and RegistrationGuard won't be able to render until DisabledGuard passes.